pkg/kubelet/kubelet_test.go

```diff
@@ -1898,6 +1899,58 @@ func TestDoesNotDeletePodDirsIfContainerIsRunning(t *testing.T) {
 	syncAndVerifyPodDir(t, testKubelet, pods, []*v1.Pod{apiPod}, false)
 }
 
+func TestFastStatusUpdateOnce(t *testing.T) {
+	testKubelet := newTestKubelet(t, false /* controllerAttachDetachEnabled */)
+	defer testKubelet.Cleanup()
+	kubelet := testKubelet.kubelet
+	kubeClient := testKubelet.fakeKubeClient
+
+	node := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: string(kubelet.nodeName),
+		},
+		Status: v1.NodeStatus{
+			Conditions: []v1.NodeCondition{
+				{
+					Type:    v1.NodeReady,
+					Status:  v1.ConditionFalse,
+					Reason:  "NotReady",
+					Message: "Node not ready",
+				},
+			},
+		},
+	}
+
+	nodeLister := testNodeLister{[]*v1.Node{node.DeepCopy()}}
+
+	kubeClient.AddReactor("patch", "nodes", func(action core.Action) (bool, runtime.Object, error) {
+		nodeLister.nodes[0].Status.Conditions[0].Status = v1.ConditionTrue
```

Comment:
can we make the first patch to fail ?

if you inject a `fake.Clock` in the kubelet we can also test that it always fails